### PR TITLE
Fix serialization scope options

### PR DESCRIPTION
**Pull request by bf4**
Fixes https://github.com/rails-api/active_model_serializers/issues/1509

### CODE REVIEW

File: test/action_controller/serialization_scope_name_test.rb [hunk 1]
```diff
@@ -1,6 +1,17 @@
...
 module SerializationScopeTesting
+  def self.undef_serializer_dynamic_scope_methods
+    [PostSerializer, PostViewContextSerializer]. each do |serializer_class|
+      instance_method_cache = serializer_class.instance_variable_get(:@_serializer_instance_methods)
+      class_method_cache    = serializer_class.class_variable_get(:@@_serializer_instance_methods)[serializer_class]
+      [:view_context, :current_user].each do |scope_method|
+        serializer_class.send(:undef_method, scope_method) if serializer_class.method_defined?(scope_method)
+        instance_method_cache && instance_method_cache.delete(scope_method)
+        class_method_cache && class_method_cache.delete(scope_method)
+      end
```
Comment (bf4) [inline test/action_controller/serialization_scope_name_test.rb]: :scream:

File: lib/active_model/serializer.rb [hunk 2]
```diff
@@ -96,16 +96,18 @@ def self.get_serializer_for(klass)
...
     end
 
-    # TODO: Fix load-order failures when different serializer instances define different
-    # scope methods
-    def self._serializer_instance_methods
-      @_serializer_instance_methods ||= (public_instance_methods - Object.public_instance_methods).to_set
+    def self._serializer_instance_method_defined?(name)
+      @_serializer_instance_methods ||= (ActiveModel::Serializer.public_instance_methods - Object.public_instance_methods).to_set
+      @_serializer_instance_methods.include?(name) ||
+        @@_serializer_instance_methods[self].include?(name)
```
Comment (bf4) [inline lib/active_model/serializer.rb]: it just works :tm:

File: test/action_controller/serialization_scope_name_test.rb (hunk 1 above)
Comment (remear) [inline test/action_controller/serialization_scope_name_test.rb]: woah

**Comment by remear** [comment]
Looks good to me.

**Pull request merged**